### PR TITLE
rename and moving files between feature/workspace layers

### PR DESCRIPTION
**Pull request by heejaechang**
only real change is having ICompilerDiagnosticAnalyzer interface which can either have inproc implementation or out of proc implementation.

inproc is needed since diagnostics are in feature layer and one who uses feature layer out side of VS host need an implementation.

...

commit with actual change - 
https://github.com/dotnet/roslyn/pull/12775/commits/8f4acd149766dfa686160532a7fe06cf503ffd38

### CODE REVIEW

**Comment by heejaechang** [comment]
@mavasani @dotnet/roslyn-ide @dotnet/roslyn-analysis can you take a look? Manish, all changes are diagnostic related. all most all of them is just rename and moving files around.

File: src/Features/Core/Portable/Diagnostics/EngineV2/ICompilerDiagnosticAnalyzer.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Workspaces.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.Diagnostics.EngineV2
+{
+    internal interface ICompilerDiagnosticAnalyzer : IWorkspaceService
```
Comment (mavasani) [inline src/Features/Core/Portable/Diagnostics/EngineV2/ICompilerDiagnosticAnalyzer.cs]: Name is confusing with [CompilerDiagnosticAnalyzer](http://source.roslyn.io/Microsoft.CodeAnalysis/R/39619341c8ee7e95.html) that lies in the compiler layer for reporting pure compiler/build diagnostics. Can we instead rename this to something like IDiagnosticAnalyzerDriver and add a comment that is the driver for executing pure diagnostic analyzers (i.e. need no workspace context) using CompilationWithAnalyzers?

File: src/Features/Core/Portable/Diagnostics/EngineV2/InProcCompilerDiagnosticAnalyzer.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Diagnostics.Telemetry;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Workspaces.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.Diagnostics.EngineV2
+{
+    [ExportWorkspaceService(typeof(ICompilerDiagnosticAnalyzer)), Shared]
+    internal class InProcCompilerDiagnosticAnalyzer : ICompilerDiagnosticAnalyzer
```
Comment (mavasani) [inline src/Features/Core/Portable/Diagnostics/EngineV2/InProcCompilerDiagnosticAnalyzer.cs]: same here - InProcDiagnosticAnalyzerDriver?

**Comment by mavasani** [comment]
Just a minor naming related suggestion, otherwise 👍 assuming its all just file moves :)

File: src/Features/Core/Portable/Diagnostics/EngineV2/ICompilerDiagnosticAnalyzer.cs (hunk 1 above)
Comment (heejaechang) [inline src/Features/Core/Portable/Diagnostics/EngineV2/ICompilerDiagnosticAnalyzer.cs]: sure

File: src/Features/Core/Portable/Diagnostics/EngineV2/InProcCompilerDiagnosticAnalyzer.cs (hunk 2 above)
Comment (heejaechang) [inline src/Features/Core/Portable/Diagnostics/EngineV2/InProcCompilerDiagnosticAnalyzer.cs]: sure

**Comment by heejaechang** [comment]
@mavasani  addressed the feedback.

**Comment by heejaechang** [comment]
thsi PR is merged to this
https://github.com/dotnet/roslyn/pull/12774

**Pull request closed without merging**